### PR TITLE
Move FingerprintPatcher out of Internal namespace

### PR DESCRIPTION
**Pull request by lucaswerkmeister**
This class is needed to write patchers for non-Item/Property entities with fingerprints, so let’s make it public. Also, like other patchers in the library it now accepts any Diff, not just an EntityDiff; this makes it more useful to patch fingerprints not connected to proper entities, and also reduces the dependency upon EntityDiff’s label / description / aliases getters, which are already marked with a FIXME because not all entities have labels, descriptions or aliases.

---

Note: add `-M42` to Git commands (`status`, `diff` etc.) to make it recognize that `FingerprintPatcher.php` was moved, not removed and added anew.

Context is [I2e68f1a6dd](https://gerrit.wikimedia.org/r/487150) and [I8ceb9e6005](https://gerrit.wikimedia.org/r/487152).

### CODE REVIEW

File: src/Diff/FingerprintPatcher.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+namespace Wikibase\DataModel\Services\Diff;
+
+use Diff\DiffOp\Diff\Diff;
+use Diff\Patcher\PatcherException;
+use Wikibase\DataModel\Services\Diff\Internal\AliasGroupListPatcher;
+use Wikibase\DataModel\Term\Fingerprint;
+
+/**
+ * @since 1.0
```
Comment (thiemowmde) [inline src/Diff/FingerprintPatcher.php]: I think this needs to say `@since 3.13` now, or be removed.

File: src/Diff/FingerprintPatcher.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	 */
+	private $aliasGroupListPatcher;
+
+	public function __construct() {
+		$this->termListPatcher = new TermListPatcher();
+		$this->aliasGroupListPatcher = new AliasGroupListPatcher();
+	}
+
+	/**
+	 * @since 1.0
```
Comment (thiemowmde) [inline src/Diff/FingerprintPatcher.php]: Can you remove this `@since`?

**Review by thiemowmde — APPROVED** [review]
Good for me.

File: src/Diff/FingerprintPatcher.php (hunk 1 above)
Comment (lucaswerkmeister) [reply src/Diff/FingerprintPatcher.php]: Yeah, I wasn’t sure what do to about these… I guess 3.13 makes more sense, yeah.

**Comment by JeroenDeDauw** [comment]
This chance looks fine to me locally, though as per my email, I don't think it is a good idea to use this in Wikibase Schema.

**Comment by mariushoch** [comment]
@lucaswerkmeister I guess this can be closed now?

**Comment by lucaswerkmeister** [comment]
I guess so, though no one has objected to this patch itself…

**Pull request closed without merging**